src/components/follow-link/follow-link.test.tsx

```diff
@@ -0,0 +1,185 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import FollowLink from '.';
+import { Tag } from '../../interfaces/tag';
+
+import * as nextAuth from 'next-auth/react';
+import * as nextRouter from 'next/router';
+
+// @ts-expect-error hacky way to mock named exports
+nextRouter.useRouter = jest.fn().mockImplementation(() => ({ asPath: '' }));
```

Comment:
can this be accomplished via
```
const mockUseRouter = jest.fn(() => ({ asPath: '' }))
jest.mock('next/router', () => ({
    __esModule: true,
    ...jest.requireActual('next/router'),
    useRouter: mockUseRouter
}))
```
? might remove the need for a `// @ts-expect-error`

Reply:
Ah yeah that seems a lot better, will do this

Reply:
@max-melo I fixed this for all the instances where it mocks for the whole test file, but I couldn't get it to work with different implementations for each test, so I left those as is. If you know a good way to do this for multiple implementations let me know. I tried `doMock`, which is [what the docs seem to suggest](https://jestjs.io/docs/jest-object#jestdomockmodulename-factory-options), but it didn't have the desired behavior.

Reply:
If I remember correctly you should be able to pass in a variable into the mock and then use `mockImplementationOnce` in each test. Just have to make sure that when mocking, you're returning a function that calls the mock instead of the mock itself or you'll get errors (something I forgot in my original comment)

```
const mockUseSession = jest.fn(())
jest.mock('next-auth/react', () => ({
    __esModule: true,
    ...jest.requireActual('next-auth/react'),
    useSession: (...args) => mockUseSession(...args)
}))
...
test('...', () => {
    mockUseSession.mockImplementationOnce(() => ({ ... }))
    ...
})
```
I think this should work, but if not I'm fine with leaving the original implementation

Reply:
Ahhh ok yep I tried the exact thing you're talking about, tried to return the mock function itself: 
```useSession: mockUseSession```

 I have it how you have it now and it seems to be working, thank you!